shaptools/netweaver.py

```diff
@@ -165,6 +169,49 @@ def is_installed(self, sap_instance=None):
             raise ValueError('provided sap instance type is not valid: {}'.format(sap_instance))
         return state
 
+    @staticmethod
+    def _get_ascs_ensa_version(processes):
+        """
+        Get ASCS ENSA version
+        """
+        if shell.find_pattern(r'enserver, EnqueueServer,.*', processes.output):
+            return 1
+        elif shell.find_pattern(r'enq_server, Enqueue Server 2,.*', processes.output):
+            return 2
+        raise ValueError('ASCS not installed or found')
+
+    @staticmethod
+    def _get_ers_ensa_version(processes):
+        """
+        Get ERS ENSA version
+        """
+        if shell.find_pattern(r'enrepserver, EnqueueReplicator,.*', processes.output):
+            return 1
+        elif shell.find_pattern(r'enq_replicator, Enqueue Replicator 2,.*', processes.output):
+            return 2
+        raise ValueError('ERS not installed or found')
+
+    def get_ensa_version(self, sap_instance):
+        """
+        Get currently installed ENSA version
+
+        Args:
+            sap_instance (str): SAP instance type. Available options: ascs, ers
+
+        Returns:
+            int: Returns the ENSA version number
+
+        Raises:
+            ValueError: ENSA system is not installed or found properly
+        """
+
+        processes = self.get_process_list(False)
```

Comment:
When we are getting the versions, should we not return errors about sapcontrol cmd failure.
Like do we not need to raise error in case of `exception=True`

Reply:
Ok. Let's put `exception=True`